refnx/reduce/batchreduction.py

```diff
@@ -251,33 +358,61 @@ class BatchReducer:
 
         reduce name scale refl1 refl2 refl3 dir1 dir2 dir3
 
-    Only rows where the value of the `reduce` column is 1 will be processed.
+    Only rows where the value of the `reduce` column is 1 and where the sample
+    name is set will be processed.
     """
 
-    def __init__(self, filename, data_folder=None, verbose=True, **kwds):
+    def __init__(self, filename=None, data_folder=None, verbose=True,
+                 persistent=True, configfile=None, **kwds):
         """
         Create a batch reducer using metadata from a spreadsheet
 
         Parameters
         ----------
-        filename : str
+        filename : str, optional
             The filename of the spreadsheet to be used. Must be readable by
-            `pandas.read_excel` (`.xls` and `.xlsx` files).
+            `pandas.read_excel` (`.xls` and `.xlsx` files). If not specified,
```

Comment:
Does pandas use xlrd to read Excel files? If not then I can't remember why xlrd is required as a dependency

Reply:
pandas does use xlrd/xlwt for reading/writing excel -- although these are both 'optional' dependencies of pandas